libbeat/esleg/eslegclient/enc.go

```diff
@@ -161,15 +191,19 @@ func NewGzipEncoder(level int, buf *bytes.Buffer, escapeHTML bool) (*gzipEncoder
 	if err != nil {
 		return nil, err
 	}
-
-	g := &gzipEncoder{buf: buf, gzip: w, escapeHTML: escapeHTML}
+	g := &gzipEncoder{
+		buf:        buf,
+		gzip:       w,
```

Comment:
is it really necessary? as `counter` is wrapping it, shouldn't all operation that were done on `gzip` now be done on `counter`?

as far as I can see there is just `Close()`, `Reset()` and `Flush()`. I believe `writerWithCounter` should implement them and proxy them to the underlying writer.

you might need an interface, but I believe it's better than having `counter` as the actual writer to be used and some methods on `gzipEncoder` changing the underlying writer of `counter`

Reply:
We already do the same with the buffer, I followed the existing pattern.

I don't see how adding 3 more function implementations that just pass the calls through is any better than this.